### ISSUE TITLE
[Accessibility]: Site admin sidebar: Difficult to use on small screens

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35867]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34430

### Problem description

<img width="412" alt="image" src="https://user-images.githubusercontent.com/9516420/169831749-d2a9afc0-1422-4173-9006-b0fd9b0b4327.png">

Difficult to see content because the sidebar takes up most of the screen. 

### Expected behavior

We should collapse the sidebar on small screens, and support opening/closing it with a toggle

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
